pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -0,0 +1,117 @@
+    const block = client.blocks[volume.path];
+    const mount_points = fsys.MountPoints.map(decode_filename);
+
+    const mount_point = get_fstab_config_with_client(client, block, false, subvol)[1];
+    const is_mounted = mount_points.indexOf(mount_point) >= 0;
+    return [mount_point, is_mounted];
+}
+
+function subvolume_unmount(volume, subvol) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "unmount", null, subvol);
+}
+
+function subvolume_mount(volume, subvol) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", null, subvol);
+}
+
+export function make_btrfs_subvolume_page(parent, volume, subvol) {
+    const [mount_point, is_mounted] = subvolume_mount_info(volume, subvol);
+    console.log(volume, subvol);
+    let location = "";
+    const use = btrfs_usage(client, volume);
+    const actions = [];
+
+    if (mount_point) {
+        location = mount_point;
+        if (is_mounted) {
+            actions.push({
+                title: _("Unmount"),
+                action: () => subvolume_unmount(volume, subvol),
+            });
+        } else {
+            actions.push({
+                title: _("Mount"),
+                action: () => subvolume_mount(volume, subvol),
+            });
+            location += " " + _("(not mounted)");
+        }
+    }
+
+    // Don't allow deleting the root subvolume
+    // TODO: check if the root subvolume is always or if this requires `btrfs subvolume get-default`?
+    if (subvol.id !== 5) {
+        // TODO in follow up
+        // actions.push({
+        //     title: _("Delete"),
+        //     action: () => subvolume_delete(volume, subvol),
+        //     danger: true,
+        // });
+    }
+
+    const card = new_card({
+        title: _("btrfs subvolume"),
+        next: null,
+        // HACK: page_location does not work well with a path that contains a Number, likely due to JSON.stringify usage?
+        page_location: ["btrfs", volume.data.uuid, String(subvol.id)],
+        page_name: subvol.pathname,
+        page_size: is_mounted && <StorageUsageBar stats={use} short />,
+        location,
+        component: BtrfsSubvolumeCard,
+        props: { volume, subvol, mount_point, is_mounted },
+        actions,
+    });
+    new_page(parent, card);
+}
+
+const BtrfsSubvolumeCard = ({ card, volume, subvol, mount_point, is_mounted }) => {
+    return (
+        <StorageCard card={card}>
+            <CardBody>
+                <DescriptionList className="pf-m-horizontal-on-sm">
+                    <StorageDescription title={_("Name")} value={subvol.pathname} />
+                    <StorageDescription title={_("ID")} value={subvol.id} />
+                    {mount_point &&
+                        <StorageDescription title={_("Mount point")} value={mount_point} />
+                    }
+                </DescriptionList>
+            </CardBody>
+        </StorageCard>);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110913-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)